spyder_kernels/customize/spydercustomize.py

```diff
@@ -583,6 +571,54 @@ def runfile(filename=None, args=None, wdir=None, namespace=None,
 builtins.runfile = runfile
 
 
+def enter_debugger(filename, continue_if_has_breakpoints, code_format):
+    """Enter debugger. Code format should be a format that accept filename."""
+    if filename is None:
+        filename = get_current_file_name()
+        if filename is None:
+            return
+
+    kernel = get_ipython().kernel
+    recursive = kernel.is_debugging()
+    if recursive:
+        parent_debugger = kernel._pdb_obj
+        sys.settrace(None)
+        globals = parent_debugger.curframe.f_globals
+        locals = parent_debugger.curframe_locals
+        # Create child debugger
+        debugger = parent_debugger.__class__(
+            completekey=parent_debugger.completekey,
+            stdin=parent_debugger.stdin, stdout=parent_debugger.stdout)
+        debugger.use_rawinput = parent_debugger.use_rawinput
+        debugger.prompt = "(%s) " % parent_debugger.prompt.strip()
+    else:
+        debugger = pdb.Pdb()
+
+    filename = debugger.canonic(filename)
+    debugger._wait_for_mainpyfile = True
+    debugger.mainpyfile = filename
+    debugger.continue_if_has_breakpoints = continue_if_has_breakpoints
+    debugger._user_requested_quit = False
+
+    if os.name == 'nt':
+        filename = filename.replace('\\', '/')
+
+    code = code_format.format(repr(filename))
+
+    if recursive:
+        # Enter recursive debugger
+        parent_debugger.message("ENTERING RECURSIVE DEBUGGER")
```

Comment:
Are you sure you want to use all caps for this message? I mean, is it really such an important message for that?

Reply:
This is to be consistant with https://github.com/ipython/ipython/blob/106a841993ee773fbe4045b8aec5cb37d6cc3eaa/IPython/core/debugger.py#L606, But I think we can drop the message.